guides/howtos/Polymorphic associations with many to many.md

```diff
@@ -401,3 +401,167 @@ end
 Overall our code looks structurally the same as `has_many` would, although at the database level our relationships are expressed with join tables.
 
 While in this guide we changed our code to cope with the parameter format required by `cast_assoc`, in [Constraints and Upserts](Constraints and Upserts.md) we drop `cast_assoc` altogether and use `put_assoc` which brings more flexibilities when working with associations.
+
+## Polymorphism with a self-referencing many_to_many
```

Comment:
I think this can be a separate guide called "Self-referencing many to many.md". What do you think? :)

Reply:
We have `## Summary` section below, so if doing this change I suggest also to change mentioned section to `### Summary`

Reply:
If this is a separate guide, then `## Summary` is correct, as all other guides. :)

Reply:
That sounds great! I'll make it a separate guide :)